sys/include/net/ng_llhdr.h

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (C) 2015 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     net
+ * @{
+ *
+ * @file
+ * @brief       Generic link layer header
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ */
+
+#ifndef LLHDR_H_
+#define LLHDR_H_
+
+#include <string.h>
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Generic link layer header
+ *
+ * The actual addresses are put into the memory following this datastructure.
+ */
+typedef struct __attribute__((packed)) {
+    uint8_t addr_len;           /**< link layer address length in byte */
```

Comment:
And if we translate 16-bit addresses in link layer: why don't we translate 48-bit addresses too and set `addr_len` implicitly to 8? There at least we have a way [standardized by the IEEE](https://standards.ieee.org/develop/regauth/tut/eui48.pdf) to translate them. And what's we there is by chance two devices in an IEEE 802.15.4 network where one has configured 00:00:00:ff:fe:00:ab:cd as its EUI-64 and one 0xabcd as its 16-bit address and they each communicate through them?